types/ext/language.d.ts

```diff
@@ -55,3 +55,20 @@ export type LanguageDescriptor<TTextPreprocessorDescriptor extends TextPreproces
 export type TextPreprocessorDescriptor = {
     [key: string]: TextPreprocessor<SafeAny>;
 };
+
+export type BidirectionalPreprocessorOptions = 'off' | 'direct' | 'inverse';
+
+export type BidirectionalConversionPreprocessor = TextPreprocessor<BidirectionalPreprocessorOptions>;
+
+export type RemoveDiacriticsPreprocessors = {
+    removeDiacritics: TextPreprocessor<boolean>;
+};
+
+export type CapitalizationPreprocessors = {
+    capitalizeFirstLetter: TextPreprocessor<boolean>;
+    decapitalize: TextPreprocessor<boolean>;
+};
+
+export type LanguageWithCapitalization = LanguageDescriptor<CapitalizationPreprocessors>;
```

Comment:
This is just to avoid type files for every language / long type annotations.

Reply:
Might be nice to have some type JSDoc.

Reply:
You mean like comments in the .d.ts files explaining the types or smth else? the types should already be used in JSDoc annotations 🤔

Reply:
Yeah, I meant comments in the .d.ts files.

Reply:
I've added just a comment on TextPreprocessor, to make up for the now deleted text in settings.html. The rest is hopefully understandable enough without comments.